src/webgl/p5.RendererGL.js

```diff
@@ -1724,54 +1726,41 @@ p5.RendererGL = class RendererGL extends Renderer {
   }
 
   /*
-   * selects which fill shader should be used based on renderer state,
-   * for use with begin/endShape and immediate vertex mode.
+   * This method will handle both image shaders and
+   * fill shaders, returning the appropriate shader
+   * depending on the current context (image or shape).
    */
-  _getImmediateFillShader() {
-    const fill = this.userFillShader;
-    if (this._useNormalMaterial) {
-      if (!fill || !fill.isNormalShader()) {
-        return this._getNormalShader();
+  _getFillShader() {
+    // If a filter like BLUR is applied, disable imageShader
+    if (this._drawingFilter) {
```

Comment:
Have you tested it? I think you need to include `this._drawingFilter` = `false` initally and then in `filter(...args){` you need to make it true so it could catch the filter() calls.